replication_impl.go

```
@@ -47,3 +93,63 @@ func (c *client) DatabaseInventory(ctx context.Context, db Database) (DatabaseIn
 	}
 	return result, nil
 }
+
+// id of this batch
+func (b batchMetadata) BatchID() string {
+	return b.ID
+}
+
+// Last tick reported by this batch
+func (b batchMetadata) Tick() Tick {
+	return b.LastTick
+}
+
+// Extend the lifetime of an existing batch on the server
+func (b batchMetadata) Extend(ctx context.Context, ttl time.Duration) error {
+	if !atomic.CompareAndSwapInt32(&b.closed, 0, 0) {
+		return nil
```

Seems this is not good. 
When closed, it should return an error, not nil